shap/explainers/_tree.py

```diff
@@ -68,53 +73,68 @@ def __init__(
         feature_perturbation="interventional",
         feature_names=None,
         approximate=False,
-        **deprecated_options,
+        # FIXME: Do not allow unexpected arguments.
+        # `link` and `linearize_link` may be passed in by Explainer, but are unused.
+        **kwargs,
```

Comment:
This `**kwargs` seems like an ugly hack, but it should at least fix the failing test for now.

Reply:
Why don't we specify link and linearize_link explicitly (and both should default to `None` so that users cannot hand over (other) arguments without effect. But this is also fine for me for now

Reply:
Yes, that seems like a better approach. In any case, after this PR I think we should try to fix the root issue of #3513